packages/deployment-service/src/resolve-production-s3-bucket-permissions.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  PutPublicAccessBlockCommand,
+  S3Client,
+  BucketCannedACL,
+  PutBucketAclCommand,
+  PutBucketOwnershipControlsCommand,
+  DeleteBucketPolicyCommand,
+  PutBucketPolicyCommand,
+} from '@aws-sdk/client-s3'
+import { OnEventHandler } from 'aws-cdk-lib/custom-resources/lib/provider-framework/types'
+
+const bucketArn = (bucketName: string): string => `arn:aws:s3:::${bucketName}`
+
+type BucketPolicyStatement = {
+  Effect: 'Allow' | 'Deny'
+  Action: string[]
+  Condition?: string[]
+  Resource: string
+  Principal: {
+    [s: string]: string
+  }
+}
+
+const resolveBucket = async (client: S3Client, bucketName: string) => {
+  await client.send(
```

Comment:
Discussed with AS: There will be a very short period during which Cloudfront won't be able to access the bucket because of the policy being dropped, however the policy HAS to be dropped to be able to alter the ACL settings. Given Cloudfront should cache the bucket contents anyway this is an acceptable, but noted risk